galley/pkg/kube/info.go

```diff
@@ -0,0 +1,68 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package kube
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	sc "k8s.io/apimachinery/pkg/runtime/schema"
+
+	"istio.io/istio/galley/pkg/runtime/resource"
+)
+
+// Info represents a known crd. It is used to drive the K8s-related machinery, and to map to
+// the proto format.
+type Info struct {
```

Comment:
`kube.Info` is pretty generic. `Schema`, `ResourceSpec`, etc. would be more descriptive.